### PR TITLE
Teach TwistedTransport to follow HTTP 301 redirects.

### PR DESCRIPTION
**Pull request by tjwitz**

### CODE REVIEW

File: suds/transport/twisted_transport.py [hunk 1]
```diff
@@ -186,6 +186,12 @@ def _request(self, request, method):
         producer = StringProducer(request.message or "")
         response = yield agent.request(method, url, headers, producer)
 
+        # If a 301 is returned, attempt to follow the first link specified in the "Location" header.
+        if 301 == response.code and response.headers.hasHeader("Location"):
```
Comment (aronbierbaum) [inline suds/transport/twisted_transport.py]: nitpick: `http.MOVED_PERMANENTLY` and should we support more response codes?

**Comment by aronbierbaum** [comment]
+1

**Comment by my8bird** [comment]
P5 Checklist 
 * https://p5.tpondemand.com/entity/2995 
 
 - [ ] Release Note
 - [ ] Tests
 - [ ] Dependency versions sheet and license files updated

File: suds/transport/twisted_transport.py (hunk 1 above)
Comment (tjwitz) [inline suds/transport/twisted_transport.py]: __Done__ Used contsants from `twisted.web.http` and included 302, 303, 307.

**Pull request merged**